### PR TITLE
Fix small memory whirlpool test failure

### PR DESCRIPTION
**Pull request by paulidale**
Fix a read off the end of the input buffer when building with OPENSSL_SMALL_FOOTPRINT defined.

It looks like an off by one error in the test to see if one or two bytes need to be read.  This warrants confirmation of course.

### CODE REVIEW

**Comment by pauli-oracle** [comment]
This will require the change in #3268

**Comment by richsalz** [comment]
ping @dot-asm ?

File: crypto/whrlpool/wp_dgst.c [hunk 1]
```diff
@@ -174,7 +174,7 @@ void WHIRLPOOL_BitUpdate(WHIRLPOOL_CTX *c, const void *_inp, size_t bits)
                 goto reconsider;
             } else
 #endif
-            if (bits >= 8) {
+            if (bits > 8) {
```
Comment (dot-asm) [inline crypto/whrlpool/wp_dgst.c]: There is comment in corresponding `else`, which would be appropriate to adjust. As "remaining less than or 8 bits".

Comment (pauli-oracle) [inline crypto/whrlpool/wp_dgst.c]: Done.

**Comment by mattcaswell** [comment]
Ping @richsalz. Can this be pushed?

**Comment by richsalz** [comment]
sure, will do it now.

**Comment by richsalz** [comment]
pushed to all three branches as indicated above.  thanks for the nudge.  CI should be clean now :)

**Pull request closed without merging**